tests/deployment_test.go

```diff
@@ -24,10 +24,316 @@
+	taskQueue := &taskqueuepb.TaskQueue{Name: "deployment-test", Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	workerDeployment := &deploymentpb.Deployment{
+		SeriesName: seriesName,
+		BuildId:    buildID,
+	}
+
+	errChan := make(chan error)
+	defer close(errChan)
+
+	workflowID := deployment.GenerateDeploymentSeriesWorkflowID(seriesName)
+	query := fmt.Sprintf("WorkflowId = '%s'", workflowID)
+	notFoundErr := fmt.Sprintf("workflow not found for ID: %s", workflowID)
+
+	// GetCurrentDeployment on a non-existing series returns an error
+	resp, err := d.FrontendClient().GetCurrentDeployment(ctx, &workflowservice.GetCurrentDeploymentRequest{
+		Namespace:  d.Namespace(),
+		SeriesName: seriesName,
+	})
+	d.Error(err)
+	d.Equal(err.Error(), notFoundErr)
+	d.Nil(resp)
+
+	// Starting a deployment workflow
+	go func() {
+		d.startDeploymentWorkflows(ctx, taskQueue, workerDeployment, errChan)
+	}()
+
+	// Verify the existence of a deployment series
+	d.EventuallyWithT(func(t *assert.CollectT) {
+		a := assert.New(t)
+
+		resp, err := d.FrontendClient().CountWorkflowExecutions(ctx, &workflowservice.CountWorkflowExecutionsRequest{
+			Namespace: d.Namespace(),
+			Query:     query,
+		})
+		a.NoError(err)
+		a.Equal(int64(1), resp.GetCount())
+	}, time.Second*5, time.Millisecond*200)
+
+	// Fetch series workflow's current deployment - will be nil since we haven't set it
+	resp, err = d.FrontendClient().GetCurrentDeployment(ctx, &workflowservice.GetCurrentDeploymentRequest{
+		Namespace:  d.Namespace(),
+		SeriesName: seriesName,
+	})
+	d.NoError(err)
+	d.Nil(resp.GetCurrentDeploymentInfo())
+
+	// todo (Shivam) - cancel if pollers are still awake
+	<-ctx.Done()
+	select {
+	case err := <-errChan:
+		d.Fail("Expected error channel to be empty but got error %w", err)
+	default:
+	}
+}
+
+// addDeploymentsAndVerifyListDeployments does the following:
+// - starts deployment workflow(s)
+// - makes a ListDeployment request (with and without seriesFilter)
+// - verifies that all the expected deployments eventually appear in the list.
+func (d *DeploymentSuite) addDeploymentsAndVerifyListDeployments(seriesName string, buildID string, withSeriesFilter bool) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	// todo (shivam) - fails when seriesName has backslashes
+	taskQueue := &taskqueuepb.TaskQueue{Name: "deployment-test", Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	workerDeployment := &deploymentpb.Deployment{
+		SeriesName: seriesName,
+		BuildId:    buildID,
+	}
+
+	// Starting a deployment workflow
+	errChan := make(chan error)
+	defer close(errChan)
+
+	// Starting a deployment workflow
+	go func() {
+		d.startDeploymentWorkflows(ctx, taskQueue, workerDeployment, errChan)
+	}()
+
+	request := &workflowservice.ListDeploymentsRequest{
+		Namespace: d.Namespace(),
+	}
+
+	if withSeriesFilter {
+		request.SeriesName = seriesName
+		d.validateListDeploymentsResponse_WithSeriesFilter(ctx, request, workerDeployment)
+	} else {
```

Comment:
thanks to you @ShahabT , I realized I was inputting `true` instead of `false` in my test without the series filter 🥲

quickly realized we need different validation logic and just did that - we basically assert the number of wf executions is greater than atleast 1 for now

Reply:
hmm.. I don't think the validation logic should be different, both should verify the expected deployment is returned. if you filter with series or not.

Reply:
discussed this on-call and agreed - have made a TODO comment